### PR TITLE
New addon: Default to bitmap mode

### LINKED ISSUE #7201: New addon: default new costumes to bitmap, instead of vector
### Why this would be helpful

Saves a click each time a new costume is created

### How this addon works

> Could you make an option to default to bitmap when creating a new costume?

This was suggested twice on Discord, and also on the feedback form.

### Possible alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mybearworld**
Resolves #7201

### Changes

When the empty vector is added, an empty bitmap is added instead.

### Reason for changes

Saves a click for people who often use bitmap

### Tests

Tested in Edge.

### CODE REVIEW

**Comment by CST1229** [comment]
I think this might also trigger when importing an empty vector? Though that's extremely minor and something no one will ever do unless they're intentionally triggering the bug.

**Comment by mybearworld** [comment]
> I think this might also trigger when importing an empty vector? Though that's an extremely minor bug and something no one will ever do unless they're intentionally triggering the bug.

I'm not sure why, but when uploading the empty SVG yourself, the hash is `4b7a854341abd530a6780bfb6a68627c.svg`, which doesn't trigger the replacing.

**Comment by WorldLanguages** [comment]
> I think this might also trigger when importing an empty vector?

Even if it did, it's not a big deal. This looks like the simplest way to implement the addon.

File: addons/default-bitmap/userscript.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export default async function ({ addon, console }) {
+  const vm = addon.tab.traps.vm;
+  const _addCostume = vm.addCostume;
+  vm.addCostume = (md5, costume, ...args) => {
+    if (md5 === "cd21514d0531fdffb22204e0ec5ed84a.svg" && !addon.self.disabled) {
+      return _addCostume.call(
+        vm,
```
Comment (WorldLanguages) [inline addons/default-bitmap/userscript.js:7]: We usually just write `this` here, but in this case it's equivalent.
For a second, I thought the VM was a parameter, though. It's a bit more confusing.

Also, if you want to follow conventions, the `_addCostume` variable should be named `originalAddCostume`.

**Comment by WorldLanguages** [comment]
@mybearworld Thanks for the PR, I'll quickly review it. Prioritizing a bit, as this was directly requested through user feedback multiple times, and it's a quite simple implementation.

**Comment by WorldLanguages** [comment]
Should we extend this so that it also defaults to bitmap when creating a new sprite/backdrop with the "paint" button?

**Comment by mybearworld** [comment]
> Should we extend this so that it also defaults to bitmap when creating a new sprite/backdrop with the "paint" button?

That would make sense.

File: addons/default-bitmap/userscript.js (hunk 1 above)
Comment (mxmou) [reply addons/default-bitmap/userscript.js]: > Also, if you want to follow conventions, the `_addCostume` variable should be named `originalAddCostume`.

There's no convention - some addons use the `original` prefix, some use `old` and a few use an underscore.

Comment (WorldLanguages) [reply addons/default-bitmap/userscript.js]: yeah, it's not strictly enforced

**Comment by mybearworld** [comment]
When trapping `vm.addSprite` to make it add the bitmap instead of the vector, the bitmap gets zoomed in for some reason:
image

**Comment by WorldLanguages** [comment]
> When trapping `vm.addSprite` to make it add the bitmap instead of the vector, the bitmap gets zoomed in for some reason

Maybe we should wait until the user goes to the Costumes tab, and just emulate a button click on the "convert to bitmap" button 🤔

**Comment by mybearworld** [comment]
That's not an instant conversion, though. It's noticable.

**Comment by WorldLanguages** [comment]
> That's not an instant conversion, though. It's noticable.

Hmm, good point.

**Comment by medlans** [comment]
This was closed too.

**Pull request closed without merging**